crates/sui-framework/packages/move-stdlib/sources/vector.move

```diff
@@ -257,4 +272,77 @@ module std::vector {
             true
         }
     }
+
+    /// Destroys two vectors `v1` and `v2` by calling `f` to each pair of elements.
+    /// Consumes both vectors.
+    /// The order of elements in the vectors is preserved.
+    public macro fun zip_do<$T1, $T2>($v1: vector<$T1>, $v2: vector<$T2>, $f: |$T1, $T2|) {
+        let v1 = $v1;
+        let mut v2 = $v2;
+        v2.reverse();
+        let len = v1.length();
+        assert!(len == v2.length());
+        v1.do!(|el1| $f(el1, v2.pop_back()));
+    }
+
+    /// Zip two vectors `v1` and `v2` by applying the function `f` to each pair of elements.
+    /// Starts from the end of the vectors.
```

Comment:
Can we update the comments on the other `zip_do*` and `zip_map*` functions to match the phrasing of `zip_do`?